src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -20,73 +19,135 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler.SemanticTokens
+            ClassificationTypeNames.RegexCharacterClass,
+            ClassificationTypeNames.RegexComment,
+            ClassificationTypeNames.RegexGrouping,
+            ClassificationTypeNames.RegexOtherEscape,
+            ClassificationTypeNames.RegexQuantifier,
+            ClassificationTypeNames.RegexSelfEscapedCharacter,
+            ClassificationTypeNames.RegexText,
+
+            ClassificationTypeNames.StringEscapeCharacter,
+            ClassificationTypeNames.Text,
+            ClassificationTypeNames.VerbatimStringLiteral,
+            ClassificationTypeNames.WhiteSpace,
+
+            // XML
+            ClassificationTypeNames.XmlDocCommentAttributeName,
+            ClassificationTypeNames.XmlDocCommentAttributeQuotes,
+            ClassificationTypeNames.XmlDocCommentAttributeValue,
+            ClassificationTypeNames.XmlDocCommentCDataSection,
+            ClassificationTypeNames.XmlDocCommentComment,
+            ClassificationTypeNames.XmlDocCommentDelimiter,
+            ClassificationTypeNames.XmlDocCommentEntityReference,
+            ClassificationTypeNames.XmlDocCommentName,
+            ClassificationTypeNames.XmlDocCommentProcessingInstruction,
+            ClassificationTypeNames.XmlDocCommentText,
+            ClassificationTypeNames.XmlLiteralAttributeName,
+            ClassificationTypeNames.XmlLiteralAttributeQuotes,
+            ClassificationTypeNames.XmlLiteralAttributeValue,
+            ClassificationTypeNames.XmlLiteralCDataSection,
+            ClassificationTypeNames.XmlLiteralComment,
+            ClassificationTypeNames.XmlLiteralDelimiter,
+            ClassificationTypeNames.XmlLiteralEmbeddedExpression,
+            ClassificationTypeNames.XmlLiteralEntityReference,
+            ClassificationTypeNames.XmlLiteralName,
+            ClassificationTypeNames.XmlLiteralProcessingInstruction,
+            ClassificationTypeNames.XmlLiteralText
+        };
+
         // TO-DO: Change this mapping once support for custom token types is added:
         // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1085998
         private static readonly Dictionary<string, string> s_classificationTypeToSemanticTokenTypeMap =
             new Dictionary<string, string>
             {
                 [ClassificationTypeNames.ClassName] = LSP.SemanticTokenTypes.Class,
                 [ClassificationTypeNames.Comment] = LSP.SemanticTokenTypes.Comment,
-                [ClassificationTypeNames.ConstantName] = LSP.SemanticTokenTypes.Variable, // TO-DO: Potentially change to custom type
-                [ClassificationTypeNames.ControlKeyword] = LSP.SemanticTokenTypes.Keyword, // TO-DO: Potentially change to custom type
-                [ClassificationTypeNames.DelegateName] = LSP.SemanticTokenTypes.Member, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.ConstantName] = ClassificationTypeNames.ConstantName, // TO-DO: Change to custom type
```

Comment:
would it be possible to create the RoslynCustomTokenTypes using the total {set of ClassificationTypeNames} - {set of classification type names mapped to LSP types}?
That way if new classification types are added they automatically get included in custom mapping.

May need to define the set of all classificationtypenames in that class, not sure if there is a better way to do it

Reply:
Is being automatically included in the custom mapping good though? Or do new classification types need a corresponding change on the VS side?

If not, is it even worth defining all of this mapping? We could just map from ClassificationTypeName to LSP.SemanticTokenType, and have the code fall back to returning the ClassificationTypeName if there isn't a match.

Reply:
Yes, new custom types do need a corresponding change on the LSP side. I feel like if new classification types are added, we may not want to automatically add them, because then we may forget to update LSP.

All of the hardcoding here is just temporary until actual custom token type support is added - I'm not quite sure what kind of changes will be needed on our side once that happens.

Reply:
I'm a little confused now. If we don't want to automatically add new custom types then isn't the previous code better, with everything in the dictionary, because then new types will fail to map?

Reply:
I believe adding a new custom type will still fail since [this](https://github.com/dotnet/roslyn/blob/4f39e6d582a3a5cbfa773bc88e951b125927d350/src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs#L245) call immediately following the dictionary lookup will throw an exception.